client/src/app/queries/applications.ts

```diff
@@ -53,6 +53,7 @@ export const useUpdateApplicationMutation = (
   const { isLoading, mutate, error } = useMutation(updateApplication, {
     onSuccess: (res) => {
       onSuccess(res);
+      queryClient.setQueryData([ApplicationsQueryKey, res.data.id], res);
```

Comment:
What does this do? Seems like this is something that would be better handled by our existing patterns (or by react query internally) rather than adding a one off implementation here.

Reply:
setQueryData updates the applications list we are currently on instantly.
This allows for avoiding waiting for a refresh after creating/editing.
As for patterns, we use react query individually per API resource which makes it difficult for testing reaons to add such optimization across all queries but why not.

Reply:
For this one, aren't we already invalidating the applications fetch query so we are going to pull in new data asynchronously anyway?

The response from an application update PUT request doesn't actually have the ID defined here
![image](https://user-images.githubusercontent.com/11218376/221915042-2d737501-d301-4475-8415-ee301f64550d.png)